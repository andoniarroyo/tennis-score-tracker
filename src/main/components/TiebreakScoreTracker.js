// @flow
// external modules
import React from 'react';

import '../styles/components/tiebreakScoreTracker.scss';

type TiebreakScoreTrackerPropsType = {
  inTiebreakMode: boolean,
  player1TieBreak: number,
  player2TieBreak: number,
}

const TiebreakScoreTracker = ({ inTiebreakMode, player1TieBreak, player2TieBreak }:
  TiebreakScoreTrackerPropsType): React.Element<*> => (
    <div className={`tiebreak-mode-${inTiebreakMode.toString()}`}>
      <div>Tiebreak:</div>
      <div className="tiebreak-content">
        <div className="score-tiebreak">
          {player1TieBreak}
        </div>
        <div className="score-tiebreak">
          {player2TieBreak}
        </div>
      </div>
    </div>
);

export default TiebreakScoreTracker;
