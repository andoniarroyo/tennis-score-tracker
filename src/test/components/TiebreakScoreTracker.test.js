import React from 'react';
import renderer from 'react-test-renderer';

import TiebreakScoreTracker from '../../main/components/TiebreakScoreTracker';

describe('<TiebreakScoreTracker />', () => {
  it('renders the tiebreak score tracker', () => {
    const props = {
      inTiebreakMode: true,
      player1TieBreak: 7,
      player2TieBreak: 8,
    };

    const tree = renderer.create(<TiebreakScoreTracker {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
