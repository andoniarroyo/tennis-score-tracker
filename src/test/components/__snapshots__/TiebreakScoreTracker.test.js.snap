// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TiebreakScoreTracker /> renders the tiebreak score tracker 1`] = `
<div
  className="tiebreak-mode-true"
>
  <div>
    Tiebreak:
  </div>
  <div
    className="tiebreak-content"
  >
    <div
      className="score-tiebreak"
    >
      7
    </div>
    <div
      className="score-tiebreak"
    >
      8
    </div>
  </div>
</div>
`;
